src/main/java/ui/factories/WindowFactory.java

```diff
@@ -1,11 +1,12 @@
 package ui.factories;
 
 import ui.windows.Window;
+import use_cases.DataAccess;
 
 public interface WindowFactory {
     /**
      * Method to be used to create a Window class upon being called.
      * @return The window that will be created.
      */
-    Window createWindow();
+    Window createWindow(DataAccess database);
```

Comment:
you dont need to propagate this change to the base interface, we can just pass the database in as a parameter to the StatsFactory object, if we need to

but otherwise dont worry about factories for now